### PR TITLE
NMS-9597 - Making the Mattermost plugin display the same format as slack

### PR DESCRIPTION
**Pull request by jemershaw**
Fix the subject formatting from `#### my-subject` to `** my subject **`

* JIRA: https://issues.opennms.org/browse/NMS-9597

### CODE REVIEW

**Comment by opennms-bamboo** [comment]
Welcome @jemershaw.

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username @jemershaw.
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome @jemershaw.

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username @jemershaw.
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

**Comment by opennms-bamboo** [comment]
Welcome @jemershaw.

We would like to thank you for your contribution.

I could not find a signed OpenNMS Contributor Agreement (OCA) assigned to the username @jemershaw.
In order to maintain some order with respect to the copyright to the code,
every contributor is asked to sign a "Contributor Agreement".

The OpenNMS Contributor Agreement can be downloaded [here](http://www.opennms.org/documentation/ContributorAgreement.pdf).
Please send a signed copy to oca@opennms.org.

If you need more information, please visit http://www.opennms.org/wiki/Contributor_Agreement

File: opennms-services/src/main/java/org/opennms/netmgt/notifd/MattermostNotificationStrategy.java [hunk 1]
```diff
@@ -80,8 +80,11 @@ protected String formatWebhookErrorResponse(int statusCode, String contents) {
 
     @Override
 	protected String decorateMessageSubject(String subject) {
-    	StringBuilder bldr = new StringBuilder("#### ");
-    	bldr.append(subject).append("\n");
+    	if ("".equals(subject)) {
+    		return "";
+    	}
+    	StringBuilder bldr = new StringBuilder("*");
```
Comment (jeffgdotorg) [inline opennms-services/src/main/java/org/opennms/netmgt/notifd/MattermostNotificationStrategy.java]: If you're going for bold, Mattermost wants two asterisks to Slack's one.
https://api.slack.com/docs/message-formatting#message_formatting
https://docs.mattermost.com/help/messaging/formatting-text.html#text-style

Comment (jemershaw) [reply opennms-services/src/main/java/org/opennms/netmgt/notifd/MattermostNotificationStrategy.java]: Good catch! Totally missed that one!

**Comment by soleger** [comment]
This branch needs to be rebased against `develop` and then force-pushed to your repository (instead of being branched from `master` and targeting our `master` branch). We use "gitflow" branch management where all current development is done in the `develop` branch and `master` is only updated to the latest stable release code.

**Comment by RangerRick** [comment]
I've cherry-picked @jemershaw's changes to a develop-based branch here: https://github.com/OpenNMS/opennms/tree/jira/NMS-9597

If it goes green I'll close this and merge the updated branch.

**Comment by RangerRick** [comment]
Closing in favor of https://github.com/OpenNMS/opennms/pull/1702

**Pull request closed without merging**